### PR TITLE
Frontend: contest public tests

### PR DESCRIPTION
**Pull request by sanyavertolet**
This PR is connected with #250 

### What's done:
- [x] Added public tests support on frontend
- [x] Added react-markdown support in order to support pretty description
- [x] Fixed a bug with wrong url in PARTICIPANTS tab
- [x] Reworked card component

### CODE REVIEW

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/contests/ContestInfoMenu.kt [hunk 1]
```diff
@@ -4,51 +4,130 @@ package com.saveourtool.save.frontend.components.basic.contests
...
+                            }
+                        }
+                    }
+                }
             }
         }
     }
 }
+
+private fun wrapTestLines(testLines: List<String>, language: String?) = "```${language ?: ""}\n${testLines.joinToString("\n")}\n```"
```
Comment (petertrr) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/contests/ContestInfoMenu.kt:133]:
```suggestion
private fun wrapTestLines(testLines: List<String>, language: String?) = """
    |```${language ?: ""}
    |${testLines.joinToString("\n")}
    |```""".trimMargin()
```
I think it looks more clear

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/contests/ContestInfoMenu.kt [hunk 2]
```diff
@@ -4,51 +4,130 @@ package com.saveourtool.save.frontend.components.basic.contests
...
+import com.saveourtool.save.test.TestFilesContent
+
 import csstype.ClassName
+import org.w3c.fetch.Headers
 import react.*
 import react.dom.html.ReactHTML.div
 
+import kotlinx.js.jso
+
+private val card = cardComponent(hasBg = true, isPaddingBottomNull = true)
```
Comment (petertrr) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/contests/ContestInfoMenu.kt:18]: Please rename it to be more descriptive

**Review by petertrr — APPROVED** [review]
LGTM, but having some screenshots would be good

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/contests/ContestInfoMenu.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/contests/ContestInfoMenu.kt:133]: It does, thanks

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/contests/ContestInfoMenu.kt (hunk 2 above)
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/contests/ContestInfoMenu.kt]: Oh, sure, thx

**Pull request merged**